cpp/CMakeLists.txt

```diff
@@ -213,8 +213,8 @@ add_library(cudf
     src/groupby/sort/sort_helper.cu
     src/hash/hashing.cu
     src/interop/dlpack.cpp
-    src/interop/from_arrow.cpp
-    src/interop/to_arrow.cpp
+    src/interop/from_arrow.cu
```

Comment:
If it's only decimal code that requires NVCC compilation, suggest keeping everything else in the old .cpp files, and putting only the CUDA/thrust-using functions in `[to|from]_arrrow.cu`. This should be better for compilation time (reduces NVCC bottleneck, improves compilation concurrency)

Reply:
@harrism is this something we really want to do? My first attempt was at just isolating the `operator()` specialization for `decimal64` to a `.cu` and then converting the old file back to `.cpp`. This necessitates removing the anonymous namespace that the `dispatch_to_cudf_column` lives in and then moving that struct and all of its member function declarations to the `cudf/detail/interop.hpp` so it can be shared across the `.cu` & `.cpp` (note I don't even have this compiling yet) ...
```cpp
// Now in cudf/detail/interop.hpp
struct dispatch_to_cudf_column {
  template <typename T>
  std::unique_ptr<column> operator()(arrow::Array const& array,
                                     data_type type,
                                     bool skip_mask,
                                     rmm::cuda_stream_view stream,
                                     rmm::mr::device_memory_resource* mr);

  std::unique_ptr<rmm::device_buffer> get_mask_buffer(arrow::Array const& array,
                                                      rmm::cuda_stream_view stream,
                                                      rmm::mr::device_memory_resource* mr);

  std::unique_ptr<column> get_empty_type_column(size_type size);

  std::unique_ptr<column> get_column(arrow::Array const& array,
                                     data_type type,
                                     bool skip_mask,
                                     rmm::cuda_stream_view stream,
                                     rmm::mr::device_memory_resource* mr);
};
```

Reply:
It used to be a .cop and worked. What changed? My suggestion is to put whatever changed into the .Cu, but not everything. Sorry on mobile so can't examine the code easily.

Reply:
The problem is that "the stuff that changed" is specializations of a type-dispatched functor. Separating that into a .cu is non-trivial. This feels like premature compile time optimization.

Reply:
Basically what Jake said. If this was a standalone function or piece of functionality, it would be straight forward. However, the fact the the code lives in an anonymous namespace and is a template specialization of a type-dispatched functor, it makes it very tricky.